File: LeetCode/199. Binary Tree Right Side.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
 
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> result=new ArrayList<>();
        
        fill(root,result,0);
        return result;
    }
    
    private void fill(TreeNode node,List<Integer> result,int depth){
        if(node==null) return;
        
        if(depth==result.size()){
            result.add(node.val);
        }
        
        fill(node.right,result,depth+1);
        fill(node.left,result,depth+1);
    }
}

File: LeetCode/199. Binary Tree Right Side.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def fill(self,node,result,depth):
        if not node:
            return
        if depth==len(result):
            result.append(node.val)
            
        self.fill(node.right,result,depth+1)
        self.fill(node.left,result,depth+1)
    
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result_list=[]
        
        self.fill(root,result_list,0)
        
        return result_list
